routes/pangolin.js

```diff
@@ -72,30 +69,43 @@ async function calcAddresses() {
   return num_addresses;
 }
 
-// WIP not correct, need to get number of swaps, not number of txs
-async function calcAvg() {
-  let result = await client.request(factoryQuery)
-  let totalVolumeETH = result['pangolinFactories'][0]['totalVolumeETH']
-  let txCount = result['pangolinFactories'][0]['txCount']
-
-  let avg = totalVolumeETH / txCount
-
-  let avaxPrice = await getAvaxPrice()
+/**
+ * Helper to discover number of swap transactions
+ */
+async function getSwapsNumber(i = 0, j = 100000) {
+  let { swaps } = await client.request(swapQuery, { skip: j });
+  let is_swaps = (swaps.length) ? swaps.length : 0;
+
+  if((is_swaps < 100 && is_swaps > 0) || j === 0) {
+    return j + is_swaps;
+  } else {
+    return is_swaps ? await getSwapsNumber(j, (j + ((j-i) * 2))) : await getSwapsNumber(i, Math.floor((j - ((j-i) / 2))));
+  }
+}
 
+/**
+ * Calculate average swap transactions on USD
+ */
+async function calcAvg() {
+  let result = await client.request(factoryQuery);
+  let totalVolumeETH = result['pangolinFactories'][0]['totalVolumeETH'];
```

Comment:
@dasconnor what about here? Is this AVAX too? Or this  totalVolumeETH is in ETH?

Reply:
Yep! That's why we multiply by the price of AVAX on line 95